### ISSUE TITLE
A loaded model with PReLU does not produce expected predictions

### ISSUE THREAD

**Issue by vsharipov** [mlpack/mlpack#3113]
#### Issue description

Hello!

I have observed the following problem. If I save and load a model, it generates different predictions (see the example below).

I think that the problem is in the serialization of the PReLU layer. If I replace it with e.g. LeakyReLU, the loaded model works as expected.

Though I noticed that the mlpack version I use is quite dated, the issue could still be present. Could you please check? Thanks!

#### My environment

 * version of mlpack: 3.0.4
 * operating system: Debian 10
 * compiler: g++ 8.3.0
 * version of dependencies (Boost/Armadillo): 1.67.0/9.200.7

#### Steps to reproduce

// Compile and run the following code

#include <mlpack/core.hpp>
#include <mlpack/methods/ann/layer/layer.hpp>
#include <mlpack/methods/ann/ffn.hpp>
using namespace mlpack;
using namespace mlpack::ann;
int main()
{
  // Load the training set and testing set
  arma::mat trainData;
  data::Load("thyroid_train.csv", trainData, true);
  arma::mat testData;
  data::Load("thyroid_test.csv", testData, true);
  // Split the labels from the training set and testing set respectively
  arma::mat trainLabels = trainData.row(trainData.n_rows - 1);
  arma::mat testLabels = testData.row(testData.n_rows - 1);
  trainData.shed_row(trainData.n_rows - 1);
  testData.shed_row(testData.n_rows - 1);
  // Initialize the network
  FFN<> model;
  model.Add<Linear<> >(trainData.n_rows, 8);
  model.Add<PReLU<> >();
  model.Add<Linear<> >(8, 3);
  model.Add<LogSoftMax<> >();
  // Train the model
  model.Train(trainData, trainLabels);
  // Use the Predict method to get the predictions
  arma::mat predictionTemp;
  model.Predict(testData, predictionTemp);
  arma::mat prediction = arma::zeros<arma::mat>(1, predictionTemp.n_cols);
  // Find index of max prediction for each data point and store in "prediction"
  for (size_t i = 0; i < predictionTemp.n_cols; ++i)
  {
    // we add 1 to the max index, so that it matches the actual test labels
    prediction(i) = arma::as_scalar(arma::find(
        arma::max(predictionTemp.col(i)) == predictionTemp.col(i), 1)) + 1;
  }
  size_t correct = arma::accu(prediction == testLabels);
  double classificationError = 1 - double(correct) / testData.n_cols;
  // Print out the classification error for the testing dataset.
  std::cout << "Classification Error for the Test set, initial model: " << classificationError << std::endl;
  // Save the model
  data::Save("model.bin", "model", model, false);
  // Load the model
  FFN<> loadedModel;
  data::Load("model.bin", "model", loadedModel);
  // Compute predictions of the loaded model on the same data set
  loadedModel.Predict(testData, predictionTemp);
  for (size_t i = 0; i < predictionTemp.n_cols; ++i)
  {
    // we add 1 to the max index, so that it matches the actual test labels
    prediction(i) = arma::as_scalar(arma::find(
        arma::max(predictionTemp.col(i)) == predictionTemp.col(i), 1)) + 1;
  }
  correct = arma::accu(prediction == testLabels);
  classificationError = 1 - double(correct) / testData.n_cols;
  // Print out the classification error of the loaded model for the testing dataset
  std::cout << "Classification Error for the Test set, loaded model: " << classificationError << std::endl;
  return 0;
}

#### Expected behavior

The classification error is the same.

#### Actual behavior

The classification error is different.

**Comment by zoq** [comment]
Thanks for the report, I can see that in the reset function we overwrite the learned `alpha` with the initial value the user provided if we load the model (https://github.com/mlpack/mlpack/blob/802b6826b26f65abdf8436e22f94d503066ff908/src/mlpack/methods/ann/layer/parametric_relu_impl.hpp#L35-L36). Will open a PR to fix the issue.

**Comment by zoq** [comment]
@shubham1206agra fixed the issue in https://github.com/mlpack/mlpack/pull/3115.
